Source/JavaScriptCore/runtime/JSString.cpp

```diff
@@ -216,7 +216,7 @@ AtomString JSRopeString::resolveRopeToAtomString(JSGlobalObject* globalObject) c
     // If we resolved a string that didn't previously exist, notify the heap that we've grown.
     if (valueInternal().impl()->hasOneRef())
         vm.heap.reportExtraMemoryAllocated(valueInternal().impl()->cost());
-    return valueInternal();
+    return AtomString { valueInternal() };
```

Comment:
I think we could change this to not need to check the atom string bit at runtime, since we know this is an `AtomStringImpl`.

Reply:
Ok, will use something like:
```
auto convertToAtomString = [](const String& string) -> AtomString {
        ASSERT(!string.impl() || string.impl()->isAtom());
        return static_cast<AtomStringImpl*>(string.impl());
};
```